buildstockbatch/test/test_integration.py

```diff
@@ -0,0 +1,73 @@
+import os
+import pathlib
+import pytest
+import shutil
+
+from buildstockbatch.localdocker import LocalDockerBatch
+
+resstock_directory = pathlib.Path(
+    os.environ.get("RESSTOCK_DIR", pathlib.Path(__file__).resolve().parent.parent.parent.parent / "resstock")
+)
+resstock_required = pytest.mark.skipif(
+    not resstock_directory.exists(),
+    reason="ResStock checkout is not found"
+)
+
+
+@pytest.mark.parametrize("project_filename", [
+    resstock_directory / "project_national" / "national_baseline.yml",
+    resstock_directory / "project_national" / "national_upgrades.yml",
+    resstock_directory / "project_testing" / "testing_baseline.yml",
+    resstock_directory / "project_testing" / "testing_upgrades.yml",
+], ids=lambda x: x.stem)
+@resstock_required
+def test_resstock_local_batch(project_filename, mocker):
+    LocalDockerBatch.validate_project(str(project_filename))
+    batch = LocalDockerBatch(str(project_filename))
+
+    # Get the number of upgrades
+    n_upgrades = len(batch.cfg.get("upgrades", []))
+    # Limit the number of upgrades to 2 to reduce simulation time
+    if n_upgrades > 2:
+        batch.cfg["upgrades"] = batch.cfg["upgrades"][0:2]
+        n_upgrades = 2
+
+    # Modify the number of datapoints so we're not here all day.
+    if n_upgrades == 0:
+        n_datapoints = 4
+    else:
+        n_datapoints = 2
+    batch.cfg["sampler"]["args"]["n_datapoints"] = n_datapoints
+
+    # FIXME: Find a better way to do this
+    local_weather_file = resstock_directory.parent / "weather" / batch.cfg["weather_files_url"].split("/")[-1]
+    if local_weather_file.exists():
+        del batch.cfg["weather_files_url"]
+        batch.cfg["weather_files_path"] = str(local_weather_file)
```

Comment:
It'll check if the weather is found locally and swap that in instead. Since we downloaded it earlier to the appropriate place it should just be able to unzip it.